core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -638,7 +638,8 @@ struct CudaParallelLaunchImpl<
 
 #if defined(KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK)
       CUDA_SAFE_CALL(cudaGetLastError());
-      cuda_instance->fence();
+      cuda_instance->fence(
+          "Kokkos::Impl::launch_kernel: Debug Only Bounds Check Fence");
```

Comment:
IMHO it's negotiable if `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` is correct here instead of just `KOKKOS_ENABLE_DEBUG`. In any case, the purpose of this `fence` is to check for execution errors in the kernel just launched. Hence, I would rather say something like "Kokkos::Impl::launch_kernel: Check for execution error");".

Reply:
I want to leave "Debug Only" in there, as I want people to be able to quickly see if a fence comes from them turning on custom, non-release flags. How would you feel about

"Kokkos::Impl::launch_kernel: Debug Only Check for Execution Error"?

Reply:
Sure, that sounds reasonable.

Reply:
552bb3c